drivers/shared/executor/legacy_executor_wrapper.go

```diff
@@ -0,0 +1,198 @@
+package executor
+
+import (
+	"encoding/gob"
+	"net/rpc"
+	"os"
+	"syscall"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	cstructs "github.com/hashicorp/nomad/client/structs"
+	"github.com/hashicorp/nomad/plugins/drivers"
+	"golang.org/x/net/context"
+)
+
+// Registering these types since we have to serialize and de-serialize the Task
+// structs over the wire between drivers and the executor.
+func init() {
+	gob.Register([]interface{}{})
+	gob.Register(map[string]interface{}{})
+	gob.Register([]map[string]string{})
+	gob.Register([]map[string]int{})
+	gob.Register(syscall.Signal(0x1))
+}
+
+type legacyExecutorWrapper struct {
+	client *pre09ExecutorRPC
+	logger hclog.Logger
+}
+
+func (l *legacyExecutorWrapper) Launch(launchCmd *ExecCommand) (*ProcessState, error) {
+	panic("not implemented")
+}
+
+func (l *legacyExecutorWrapper) Wait(ctx context.Context) (*ProcessState, error) {
+	ps, err := l.client.Wait()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ProcessState{
+		Pid:      ps.Pid,
+		ExitCode: ps.ExitCode,
+		Signal:   ps.Signal,
+		Time:     ps.Time,
+	}, nil
+}
+
+func (l *legacyExecutorWrapper) Shutdown(signal string, gracePeriod time.Duration) error {
+	if err := l.client.ShutDown(); err != nil {
+		return err
+	}
+
+	if err := l.client.Exit(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *legacyExecutorWrapper) UpdateResources(*drivers.Resources) error {
+	panic("not implemented")
```

Comment:
Panics make me nervous. Could this crash the agent? If so let's make it an error instead.

Reply:
Yeah the reason I left he panic was that this should never ever be called. But we can error on the crash safety side and just return an error.